### PR TITLE
Show prices on pages without a REG_TYPES area

### PR DESCRIPTION
**Pull request by kitsuta**
Fixes https://github.com/magfest/ubersystem/issues/2433 by triggering togglePrices(); AFTER the badge_types area has been built, without depending on a click event that doesn't happen on some pages. Also tweaks when prices for the base badge are displayed, effectively turning it off for new group members (because they don't pay any money for their badge, so it's confusing to show them their badge's price) and anyone without a badge cost.

### CODE REVIEW

File: uber/templates/preregistration/attendee_donation_form.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
     {% if attendee.paid == c.NOT_PAID %}
       <h2> Badge Payment for {{ attendee.full_name }} </h2>
 
-      You've registered for {{ c.EVENT_NAME }} at a {% if attendee.overridden_price %}discounted{% endif %} price of ${{ attendee.overridden_price|default(attendee.badge_cost|round(2)) }}{% if attendee.amount_extra %} and you've also kicked in ${{ attendee.amount_extra|round(2) }}{% endif %}; your total outstanding balance is ${{ attendee.amount_unpaid|round(2) }}.
+      You've registered for {{ c.EVENT_NAME }} at a {% if attendee.overridden_price %}discounted{% endif %} price of ${{ attendee.badge_cost }}{% if attendee.amount_extra %} and you've also kicked in ${{ attendee.amount_extra|round(2) }}{% endif %}; your total outstanding balance is ${{ attendee.amount_unpaid|round(2) }}.
```
Comment (kitsuta) [inline uber/templates/preregistration/attendee_donation_form.html]: This is an unrelated fix; I noticed that this page was displaying $None for the badge price even if the attendee had a badge cost. Probably introduced by the jinja2 switchover.

File: uber/templates/preregistration/preregbase.html [hunk 2]
```diff
@@ -239,6 +238,7 @@
                         $.field('amount_extra').on('change', makeBadgeMatchExtra);
                     {% endif %}
                 }
+                togglePrices();
```
Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: The diff is annoying but this is the core change that fixes the bug. We now trigger togglePrices() after printing all the badge_types stuff on the page. The only reason we ever saw prices on the reg form was because every click event on a reg_types button also triggers togglePrices, and we automatically trigger a click event on the Attendee reg type button when we load the prereg form.

File: uber/templates/preregistration/preregbase.html [hunk 3]
```diff
@@ -79,7 +79,7 @@
                 title: '{% if c.PAGE_PATH in ["/preregistration/form", "/preregistration/post_form", "/preregistration/dealer_registration"] or undoing_extra %}Attending{% elif attendee %}{{ attendee.ribbon_and_or_badge }}{% endif %}',
                 description: 'Allows access to the convention for its duration.',
                 extra: 0,
-                price: {{ c.BADGE_PRICE }},
+                price: {{ badge_cost if badge_cost is defined else c.BADGE_PRICE }},
```
Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: This allows us to not display the attendee badge price if it's not appropriate, particularly for attendees confirming/editing a registration and attendees registering for a group.

File: uber/site_sections/preregistration.py [hunk 4]
```diff
@@ -411,7 +411,8 @@ def register_group_member(self, session, group_id, message='', **params):
             'group_id': group_id,
             'group': group,
             'attendee': attendee,
-            'affiliates': session.affiliates()
+            'affiliates': session.affiliates(),
+            'badge_cost': 0
```
Comment (kitsuta) [inline uber/site_sections/preregistration.py]: Even though group members technically have a badge_cost, we don't want it to show on the page. Their badge is (presumably) already paid for, so we don't want them to think they have to pay more on top of that if they're not kicking in extra.

File: uber/templates/preregistration/preregbase.html (hunk 2 above)
Comment (RobRuana) [inline uber/templates/preregistration/preregbase.html]: The call to `togglePrices()` has been moved inside the `if ($(BADGE_TYPES.row).size()) {` conditional. Was that intentional?

`togglePrices()` internally also manipulates `REG_TYPES`, so I would guess we would want to call `togglePrices()` even if `BADGE_TYPES` isn't used?

Comment (kitsuta) [inline uber/templates/preregistration/preregbase.html]: Ah, yeah, good point! I'll rejigger that a bit.
